# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build egl-opengl-stubs-build gcc-g++-build sdl2_image-build sdl2_mixer-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --print-requires sdl2-build | grep -q egl-opengl-stubs-build || { echo SDL2 with OpenGL support required; exit 1; }

# configure (ac-2.69; am-1.15; lt-2.4.2)
libtoolize --ltdl -c -f; aclocal; autoconf; autoheader -f; cp README.md README; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig ./configure --disable-static --with-highscore-prefix=/x11/var/games/pinball --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install src/pinball $DESTDIR/x11/bin
install -d $DESTDIR/x11/lib/pinball
install -m 644 data/professor/libModuleProfessor.la $DESTDIR/x11/lib/pinball
install data/professor/.libs/libModuleProfessor.so.0.* $DESTDIR/x11/lib/pinball/libModuleProfessor.so.0
install -m 644 data/tux/libModuleTux.la $DESTDIR/x11/lib/pinball
install data/tux/.libs/libModuleTux.so.0.* $DESTDIR/x11/lib/pinball/libModuleTux.so.0
install -d $DESTDIR/x11/share/pinball
install -m 644 data/font_34.png $DESTDIR/x11/share/pinball
install -m 644 data/splash.png $DESTDIR/x11/share/pinball
install -d $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/bump.wav $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/bumpsoft.wav $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/flip.wav $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/floor.png $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/gameover.wav $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/introp.ogg $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/nudge.wav $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/pinball.pbl $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/professor.ogg $DESTDIR/x11/share/pinball/professor
install -m 644 data/professor/professor.png $DESTDIR/x11/share/pinball/professor
install -d $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/bumphard.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/bumpsoft.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/bump.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/face2.png $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/flip.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/floor2.png $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/floor3.png $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/game.ogg $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/gameover.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/lock.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/intro.ogg $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/multiball.ogg $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/nudge.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/opentux.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/pinball.pbl $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/shoot.wav $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/tiles.png $DESTDIR/x11/share/pinball/tux
install -m 644 data/tux/up.wav $DESTDIR/x11/share/pinball/tux

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/pinball.pc << EOF
Name: pinball
Version: 20230219
Description: Pinball
Requires: $REQUIRES

exec=\\
/x11/bin/pinball \\
/x11/lib/pinball/libModuleProfessor.la \\
/x11/lib/pinball/libModuleProfessor.so.0 \\
/x11/lib/pinball/libModuleTux.la \\
/x11/lib/pinball/libModuleTux.so.0 \\
/x11/share/pinball/font_34.png \\
/x11/share/pinball/splash.png \\
/x11/share/pinball/professor/bump.wav \\
/x11/share/pinball/professor/bumpsoft.wav \\
/x11/share/pinball/professor/flip.wav \\
/x11/share/pinball/professor/floor.png \\
/x11/share/pinball/professor/gameover.wav \\
/x11/share/pinball/professor/introp.ogg \\
/x11/share/pinball/professor/nudge.wav \\
/x11/share/pinball/professor/pinball.pbl \\
/x11/share/pinball/professor/professor.ogg \\
/x11/share/pinball/professor/professor.png \\
/x11/share/pinball/tux/bumphard.wav \\
/x11/share/pinball/tux/bumpsoft.wav \\
/x11/share/pinball/tux/bump.wav \\
/x11/share/pinball/tux/face2.png \\
/x11/share/pinball/tux/flip.wav \\
/x11/share/pinball/tux/floor2.png \\
/x11/share/pinball/tux/floor3.png \\
/x11/share/pinball/tux/game.ogg \\
/x11/share/pinball/tux/gameover.wav \\
/x11/share/pinball/tux/lock.wav \\
/x11/share/pinball/tux/intro.ogg \\
/x11/share/pinball/tux/multiball.ogg \\
/x11/share/pinball/tux/nudge.wav \\
/x11/share/pinball/tux/opentux.wav \\
/x11/share/pinball/tux/pinball.pbl \\
/x11/share/pinball/tux/shoot.wav \\
/x11/share/pinball/tux/tiles.png \\
/x11/share/pinball/tux/up.wav
EOF

exit
--- pinball-20230219.orig/configure.ac
+++ pinball-20230219/configure.ac
@@ -8,7 +8,7 @@
 	[https://pinball.sf.net])
 
 AC_CONFIG_MACRO_DIRS([libltdl/m4])
-AM_CONFIG_HEADER(pinconfig.h)
+AC_CONFIG_HEADERS(pinconfig.h)
 LT_CONFIG_LTDL_DIR([libltdl])
 
 AC_CANONICAL_HOST
@@ -39,6 +39,8 @@
 LT_INIT
 LTDL_INIT
 
+PKG_PROG_PKG_CONFIG
+
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
 
@@ -122,15 +124,11 @@
 	[enable_gles=no])
 if test "x$enable_gles" = "xyes"; then
   MATHLIB="-lm"
-  AC_CHECK_HEADER(
-      [GLES/gl.h],
-      [ac_cv_search_glLoadMatrixx="none required"],
+  PKG_CHECK_MODULES(
+      [GL], [glesv1_cm],
+      [],
       [AC_MSG_ERROR(*** OpenGL ES headers not found on system!)]
   )
-  AC_ARG_WITH(gles-library,
-  	[AS_HELP_STRING([--with-gles-library=LIBRARY],[OpenGL ES library @<:@default=GLESv1_CM@:>@])],
-  	GL_LIBS=-l"$withval",
-  	GL_LIBS="-lGLESv1_CM")
   AC_DEFINE(HAVE_OPENGLES, [1], [Support for OpenGL ES])
 else
 case "$target" in
@@ -150,9 +148,9 @@
         ;;
     *)
         MATHLIB="-lm"
-        AC_CHECK_HEADER(
-            [GL/gl.h],
-            AC_SEARCH_LIBS(glVertex3f, [opengl32 GL GL2 OpenGL], [test $ac_cv_search_glVertex3f = "none required" || GL_LIBS="$ac_cv_search_gVertex3f $GL_LIBS"], [AC_MSG_ERROR(*** OpenGL library not found on system!)]),
+        PKG_CHECK_MODULES(
+            [GL], [gl],
+            [],
             [AC_MSG_ERROR(*** OpenGL headers not found on system!)]
         )
         ;;
--- pinball-20230219.orig/src/Pinball.cpp
+++ pinball-20230219/src/Pinball.cpp
@@ -11,7 +11,6 @@
 #include <fstream>
 #include <string>
 //#include <sstream>
-#include <strstream>
 #include <iostream>
 
 #include <sys/stat.h>
@@ -144,7 +143,7 @@
     return EM_MENU_NOP;
   }
   const char * getText() {
-    ostrstream stm;
+    ostringstream stm;
     stm.clear();
     string name(m_Name);
     const char * keyname = Config::getInstance()->getKeyCommonName(Config::getInstance()->getKey(name));
